lib/src/solver/result.dart

```diff
@@ -116,6 +130,40 @@ class SolveResult {
     }
   }
 
+  /// Send analytics about the package resolution.
+  void sendAnalytics(PubAnalytics analytics) {
+    ArgumentError.checkNotNull(analytics);
+
+    for (final package in packages) {
+      final source = package.source;
+      // Only send analytics for packages from pub.dev.
+      if (source is HostedSource &&
+          (runningFromTest ||
+              package.description['url'] == HostedSource.pubDevUrl)) {
+        final dependencyKind = const {
+          DependencyType.dev: 'dev',
+          DependencyType.direct: 'direct',
+          DependencyType.none: 'transitive'
+        }[_root.dependencyType(package.name)];
+        analytics.analytics.sendEvent(
+          'pub-get',
+          package.name,
+          label: package.version.toString(),
```

Comment:
Do we need to normalize the version string encoding here?

Reply:
We should probably normalize using: https://github.com/dart-lang/pub_semver/pull/50

Reply:
Done